causalml/inference/meta/tmle.py

```diff
@@ -126,7 +126,7 @@ def __repr__(self):
             self.__class__.__name__, self.model_tau.__repr__(), self.cv
         )
 
-    def estimate_ate(self, X, treatment, y, p, segment=None, return_ci=False):
```

Comment:
Ideally, it should follow the signature from `causalml.inference.meta.base.BaseLearner.estimate_ate()`. Can you update `TMLELearner` to inherit from `BaseLearner` and keep the function signatures consistent with other meta learners?

Reply:
Will keep this pull request simple. 
Will open a new one for a deeper dive into `tmle.py`